### ISSUE TITLE
Manifold-SAE under-identifies + non-PD curvature anchor on REAL OLMo activations (the #1026 recon-parity gap, mechanized)

### ISSUE THREAD

**Issue by SauersML** [SauersML/gam#1190]
## What

Running the manifold-SAE at K=8 (periodic, d_atom=2) on the **real OLMo-3-32B L25 activation fixture** (`tests/data/olmo_l25_pca64_768.npy`, 768×64 PCA, committed) exposes two concrete failures that together are the **#1026 recon-parity gap on real (not synthetic) data**. The synthetic planted-circle fixtures pass; real long-tailed activations do not.

## Reproduction

512 train / 256 held-out split of the committed fixture, then:
```python
fit = gamfit.sae_manifold_fit(X=z_train, K=8, atom_basis="periodic", d_atom=2,
                              assignment="ibp_map", n_iter=60, learning_rate=0.04, random_state=0)
```
(See `tests/test_sae_manifold_olmo_real_recon_ev.py`, the RED e2e test, and `.wager.py`.)

## Failure 1 — every atom is rank-deficient on real data
All 8 decoder atoms log:
```
[SAE-AUDIT] decoder atom 'atom_k' weighted design is rank-deficient (rank=3/5, 2 weakly-identified column(s), n=512); the Arrow-Schur ridge will regularise the deficient directions
```
The periodic d=2 atom basis (5 columns) is **rank-3 on real activations** — 2 of 5 columns are weakly identified. The ridge papers over it, but the atoms are under-determined, so reconstruction cannot earn its EV. The basis is effectively over-parameterized for the real spectrum.

## Failure 2 — non-PD curvature anchor → homotopy refused → cascade fallback
```
[#1007] curvature anchor solve failed at η=0 (SaeManifoldTerm::reml_criterion: arrow-Schur:
Schur complement Cholesky failed: non-PD pivot -1.7111... at index 29 (matrix is not positive definite)); deferring to cascade
[OUTER] SAE manifold: curvature-homotopy entry refused seed 0; deferring to the seed cascade from the pristine baseline
```
The #1007 curvature-homotopy anchor (the certified entry that replaced multistart) **fails on real data** — the Schur complement is indefinite (negative pivot) at η=0 — so the fit drops to the slow seed cascade. The fit then crawls (>3 min, still going) instead of the clean anchored entry the synthetic cases get.

## Context / the bar

Linear PCA held-out OOS EV on the same split (the parity ceiling the manifold must clear at far fewer params — manifold K=8 has `n_params=10` vs linear K=8's ~512):

| K | linear OOS EV |
|---|---|
| 1 | 0.315 |
| 2 | 0.406 |
| 4 | 0.490 |
| 8 | 0.564 |
| 16 | 0.656 |

## Why it matters

This is the real-data instance of #1026: on genuine LLM activations the curved-atom path (a) can't identify its basis and (b) can't even take the certified curvature-anchor entry. Both are concrete, code-improvable. Likely directions: a rank-revealing / lower-d atom basis adapted to the local identified rank (3, not 5); a PSD-projected or shifted curvature anchor that survives an indefinite real-data Schur complement (mirrors the #979 PSD-penalized stabilizing-shift fix); and the geometry-derived one-shot init so the fit starts in the identified subspace.

Found by running the real-data wager experiment, not synthetic. Fixture + repro committed.

**Comment by SauersML** [comment]
Superseded by the corrected literal reproduction comment below: https://github.com/SauersML/gam/issues/1190#issuecomment-4722177653

The original body was mangled by shell backtick expansion during posting.

**Comment by SauersML** [comment]
## Corrected reproduction data: exact transcript plot script on MSI

The previous comment was mangled by shell backtick expansion. This is the literal record.

I recovered the exact `/tmp/mk_clean2.py` body from the Codex transcript and ran that script on MSI/acn116 against the full banked OLMo activation tensor.

- script extracted from: `~/.codex/sessions/2026/06/16/rollout-2026-06-16T13-07-32-019ed19d-fd8f-75d2-962d-ed7adbeca181.jsonl`
- extracted script SHA-256: `a218297dc666d615809514d145ed36ea67dce5a1484c14bf6eedaee8172c42a5`
- data: `/projects/standard/REDACTED/REDACTED/olmo_data/base/activations.npy`
- output: `/projects/standard/REDACTED/REDACTED/plots/olmo_structure_clean.png`
- fresh output MD5: `72f9a3fcbe10ab422c7243a00e06ca5a`
- fresh output size: `360162` bytes

This confirms the plot was generated from the full `635 x 64 x 5120` real OLMo bank on MSI, not the GitHub-sized fixture and not a local redraw.

The real-data SAE wager still exposes the core failure mechanism tracked here: on the real OLMo PCA slice, the K=8 periodic manifold fit hits rank-deficient atom designs and a non-PD curvature-anchor Schur complement before it can compete cleanly with the linear PCA held-out EV baseline. The issue remains open until the production fit accepts the real-data anchor, avoids the slow seed cascade, and clears the held-out reconstruction bar.

**Comment by SauersML** [comment]
VERIFIED GREEN on MSI (clean builds, jobs 11146579 + 11147353). Both gate tests pass:
- terms::analytic_penalties::tests::softmax_row_fisher_metric_is_psd_gauge_null_and_derivative_matches_fd ... ok
- terms::sae::manifold::tests::olmo_real_curvature_anchor_is_positive_definite ... ok
Fix (097085b29): replaced the indefinite softmax-ENTROPY Hessian in the SAE curvature anchor with the PSD softmax Fisher-information metric G = scale·(diag(a) − a·aᵀ), so the per-row evidence Hessian (the #1007 curvature anchor) is PD by construction on real OLMo activations — the deflation no longer flattens data-relevant logit directions, and all three differentiation sites (assembly / log|H| ρ-gradient / #1006 θ-adjoint) use the identical G so value, log|H|, and adjoints stay on one branch (no desync). The entropy penalty's stationary point is unchanged (its exact gradient is unchanged; only the Newton/Laplace curvature is conditioned — same logic as the ARD max(0.0) precedent). The real-OLMo PD probe (5e7bd4ca4: K=2/160-row slice, no inner fit) asserts every per-row raw H_tt block has relative min eigenvalue ≥ −1e-8.
